### PR TITLE
Signal to update inferred schema

### PR DESCRIPTION
**Pull request by benrudolph**
@NoahCarnahan @czue this adds a signal for the inferred schema to be updated. it also refactors the inferred schema generation a little and adds more database constraints on what can be saved to the schema. 🐟 or 🏠 

These properties were "inferred" from a user case save. kind of cool:
screen shot 2016-10-28 at 3 28 37 pm

cc: @emord @nickpell

### CODE REVIEW

File: corehq/apps/callcenter/utils.py [hunk 1]
```diff
@@ -153,6 +154,13 @@ def valid_element_name(name):
         'language': commcare_user.language or '',
         'phone_number': commcare_user.phone_number or ''
     })
+
+    added_inferred_export_properties.send(
+        'UserSave',
+        domain=commcare_user.domain,
+        case_type=USERCASE_TYPE,
+        properties=fields.keys()
+    )
```
Comment (snopoke) [inline corehq/apps/callcenter/utils.py]: 👍 thanks

**Review by snopoke — COMMENTED** [review]
Looks good

**Review by NoahCarnahan — APPROVED** [review]
lgtm

File: corehq/apps/importer/tasks.py [hunk 2]
```diff
@@ -78,6 +80,14 @@ def _submit_caseblocks(caseblocks):
...
                     row_number=caseblocks[0]._id
                 )
+            else:
+                properties = set().union(*map(lambda c: set(c.dynamic_case_properties().keys()), cases))
+                added_inferred_export_properties.send(
+                    'CaseImporter',
+                    domain=domain,
+                    case_type=case_type,
+                    properties=properties,
+                )
```
Comment (benrudolph) [inline corehq/apps/importer/tasks.py]: i don't think we do. the merge will handle the duplicates and i figured it'd add unnecessary complexity to try and filter out the ones that are already in the schema

**Comment by benrudolph** [comment]
merging on approval. right now it's a django signal, but we can easily change to an async task if we think it's too much overhead

**Comment by czue** [comment]
i'm way behind on code review, but what's with the duplicate `one` and `two` properties in the screenshot? 

assume just some weird thing you were testing but wanted to ask in case there is a bug where we don't properly resolve duplicate properties or something

**Comment by benrudolph** [comment]
gah good eyes... this hasn't been deployed yet, so going to revert and figure out a solution. regular case properties get marked as `ScalarItem` which makes them not resolve with `ExportItem`.

**Comment by benrudolph** [comment]
suppose tests only help when you write them for the real scenario

**Pull request merged**